src/anndata/_io/specs/methods.py

```diff
@@ -1023,35 +1030,83 @@ def write_nullable_integer(
     dataset_kwargs: Mapping[str, Any] = MappingProxyType({}),
 ):
     g = f.require_group(k)
-    if v._mask is not None:
-        _writer.write_elem(g, "mask", v._mask, dataset_kwargs=dataset_kwargs)
+    _writer.write_elem(g, "mask", v._mask, dataset_kwargs=dataset_kwargs)
```

Comment:
> Can anyone think of any reason that always writing the mask might break some things?

Not only I can’t think of a reason, but `self._mask` is typed as `np.ndarray` from the first commit of the extension array types. No idea why anyone thought it could be `None`.

Reply:
> While we're here, we should stop relying on the private API IMO.

I’d rather do that in another PR. `_data` is also a private API.